rasa_sdk/forms.py

```diff
@@ -844,57 +880,20 @@ async def next_requested_slot(
             If the `SlotSet` event sets `requested_slot` to `None`, the form will be
             deactivated.
         """
-        required_slots = await self.required_slots(dispatcher, tracker, domain)
-        missing_slots = await self.missing_slots(dispatcher, tracker, domain)
-
-        if required_slots is None and not missing_slots:
-            # It seems like `required_slots` wasn't overridden. In this case we
-            # leave the deactivation of the form to the `FormAction` within
-            # Rasa Open Source
-            return None
-
-        missing_slots = await self.missing_slots(dispatcher, tracker, domain)
-        if not missing_slots:
-            # No more missing_slots slots. Form can be deactivated.
-            return SlotSet(REQUESTED_SLOT, None)
-
-        # request next slot
-        return SlotSet(REQUESTED_SLOT, missing_slots[0])
-
-    async def missing_slots(
-        self,
-        dispatcher: "CollectingDispatcher",
-        tracker: "Tracker",
-        domain: "DomainDict",
-    ) -> List[Text]:
-        """Returns slots which still need to be filled.
-
-        Args:
-            dispatcher: the dispatcher which is used to
-                send messages back to the user.
-            tracker: the conversation tracker for the current user.
-            domain: the bot's domain.
-
-        Returns:
-            The slots which aren't already filled.
-        """
-        required_slots = await self.required_slots(dispatcher, tracker, domain)
-        if required_slots is None:
-            return []
-
-        slots_mapped_in_domain = self.slots_mapped_in_domain(domain)
-
-        all_required_slots = required_slots + [
-            domain_slot
-            for domain_slot in slots_mapped_in_domain
-            if domain_slot not in required_slots
-        ]
+        required_slots = await self.required_slots(
+            self.slots_mapped_in_domain(domain), dispatcher, tracker, domain
+        )
+        if required_slots and required_slots == self.slots_mapped_in_domain(domain):
```

Comment:
count ordering be an issue here? Maybe a `set` is safer.

Reply:
Ordering matters here. The user might want to make a single change to the form: changing the order in which the slots are requested.